src/System.Net.WebHeaderCollection/src/System/Net/HeaderInfoTable.cs

```diff
@@ -10,64 +10,137 @@ namespace System.Net
     internal class HeaderInfoTable
     {
         private static Hashtable HeaderHashTable;
-        private static HeaderInfo UnknownHeaderInfo = new HeaderInfo(string.Empty, false, false, false);
+        private static HeaderInfo UnknownHeaderInfo = new HeaderInfo(string.Empty, false, false, false, SingleParser);
```

Comment:
`SingleParser` needs to be defined above the `UnknownHeaderInfo` static field. As it is right now, it will be passed-in to the `HeaderInfo` constructor as null.